src/v/cluster/partition_recovery_manager.cc

```diff
@@ -604,20 +611,27 @@ partition_downloader::find_recovery_material() {
     vlog(
       _ctxlog.info,
       "Downloading partition manifest {}",
```

Comment:
> Note, while the source cluster uuid of the path providers will
ultimately be the same, this is a different path provider than what will
be used at runtime by the partition. The latter is owned by the archival
metadata STM and its lifecycle is therefore tied to the partition.

I could imagine there a strategic place to put an explanation like this into the code.

Reply:
Will do in a follow-up

Reply:
Done